examples/with-remix-thirdpartyemailpassword/app/routes/_index.tsx

```diff
@@ -0,0 +1,203 @@
+import {
+  CelebrateIcon,
+  SeparatorLine,
+  BlogsIcon,
+  GuideIcon,
+  SignOutIcon,
+} from "../../assets/images";
+import { signOut } from "supertokens-auth-react/recipe/thirdpartyemailpassword/index.js";
+import { recipeDetails } from "../config/frontend";
+import SuperTokens from "supertokens-auth-react";
+import { LoaderFunctionArgs, redirect } from "@remix-run/node";
+import {
+  PreParsedRequest,
+  CollectingResponse,
+} from "supertokens-node/framework/custom/index.js";
+import Session from "supertokens-node/lib/build/recipe/session/index.js";
+import { HTTPMethod } from "supertokens-node/types";
+import { useLoaderData } from "@remix-run/react";
+import { ExtendedSession, SessionDataForUI } from "../lib/superTokensTypes";
+import { SessionContainerInterface } from "supertokens-node/lib/build/recipe/session/types";
+
+function getCookieFromRequest(request: Request) {
+  const cookies: Record<string, string> = {};
+  const cookieHeader = request.headers.get("Cookie");
+  if (cookieHeader) {
+    const cookieStrings = cookieHeader.split(";");
+    for (const cookieString of cookieStrings) {
+      const [name, value] = cookieString.trim().split("=");
+      cookies[name] = value;
+    }
+  }
+  return cookies;
+}
+
+function getQueryFromRequest(request: Request) {
+  const query: Record<string, string> = {};
+  const url = new URL(request.url);
+  const searchParams = url.searchParams;
+  searchParams.forEach((value, key) => {
+    query[key] = value;
+  });
+  return query;
+}
+
+function createPreParsedRequest(request: Request): PreParsedRequest {
```

Comment:
this function can go in a file like supertokensHelper, and then you don't need it here, nor do you need it in the middleware file

Reply:
Moved this function into the separate file as suggested.